tests/rembert/test_modeling_tf_rembert.py

```diff
@@ -168,7 +168,55 @@ def create_and_check_model(
 
         self.parent.assertEqual(result.last_hidden_state.shape, (self.batch_size, self.seq_length, self.hidden_size))
 
-    def create_and_check_lm_head(
+    def create_and_check_causal_lm_base_model(
+        self, config, input_ids, token_type_ids, input_mask, sequence_labels, token_labels, choice_labels
+    ):
+        config.is_decoder = True
+
+        model = TFRemBertModel(config=config)
+        inputs = {"input_ids": input_ids, "attention_mask": input_mask, "token_type_ids": token_type_ids}
+        result = model(inputs)
+
+        inputs = [input_ids, input_mask]
+        result = model(inputs)
+
+        result = model(input_ids)
+
+        self.parent.assertEqual(result.last_hidden_state.shape, (self.batch_size, self.seq_length, self.hidden_size))
+
+    def create_and_check_model_as_decoder(
+        self,
+        config,
+        input_ids,
+        token_type_ids,
+        input_mask,
+        sequence_labels,
+        token_labels,
+        choice_labels,
+        encoder_hidden_states,
+        encoder_attention_mask,
+    ):
+        config.add_cross_attention = True
+
+        model = TFRemBertModel(config=config)
+        inputs = {
+            "input_ids": input_ids,
+            "attention_mask": input_mask,
+            "token_type_ids": token_type_ids,
+            "encoder_hidden_states": encoder_hidden_states,
+            "encoder_attention_mask": encoder_attention_mask,
+        }
+        result = model(inputs)
```

Comment:
Is the idea here to test that both `inputs` and `input_ids` work correctly - ok for me. What do you think @Rocketknight1 ?

Reply:
Yes. I copied the tests for TFBertModel on current master, where it has

        inputs = {
            "input_ids": input_ids,
            "attention_mask": input_mask,
            "token_type_ids": token_type_ids
        } --> Test *dict*
        result = model(inputs)
        inputs = [input_ids, input_mask] --> Test *list*
        result = model(inputs)
        result = model(input_ids) --> Test *tensor*

Reply:
I don't fully understand this code - it seems like `result` is overwritten twice before only testing the final version of it?

Reply:
Just to make sure it run, and use only one result to check shapes and/or values.